v2/pkg/genruntime/secrets.go

```diff
@@ -65,6 +65,56 @@ func (s NamespacedSecretReference) String() string {
 	return fmt.Sprintf("Namespace: %q, %s", s.Namespace, s.SecretReference)
 }
 
+// SecretCollectionReference is a reference to a Kubernetes secret in the same namespace as
+// the resource it is on.
+// +kubebuilder:object:generate=true
+type SecretCollectionReference struct { // TODO: Should we be calling this SecretMap instead?
```

Comment:
Do people have a better name for this type? I'm open to it.

Reply:
I'm not a fan of this name, it can be parsed too many ways.
* A secret (confidential) reference to a collection
* A reference to a collection of secrets
* A collection of references to secrets

Maybe `SecretsReference` ?

Reply:
I don't really like the name I have chosen here either, but I am not sure that `SecretsReference` is better...
`SecretsReference` is so similar to `SecretReference` (which is a type we already have).

Is `SecretMapReference` any better than `SecretCollectionReference`?